### ISSUE TITLE
Semicolon in the end of lerp.bm example output

### ISSUE THREAD

**Issue by kolumb** [tsoding/bm#147]
For some reason instead of zero I get a semicolon as a fraction of the last number.
```console
c:\bm>.\build\bin\bme -i .\build\examples\lerp.bm
69.0
104.1
139.2
174.3
209.4
244.5
279.6
314.7
349.8
384.9
419.:
```
---
I tried to narrow down required conditions to reproduce.

* It seems that **`n`** has to be equal to 10.0

* **`b`** has to have .0 in the end. The rest of digits can be anything. I checked several examples of `b` and found corresponding biggest value of `a` that produces a semicolon.

* **`a`** can be anything starting from -INF to specified here number:
```
push 34463.864369146995159098 ;; a
push 100000.0                 ;; b

push 1807.9999 ;; a
push 10000.0   ;; b

push 487.9999 ;; a
push 1000.0   ;; b

push 17.9999 ;; a
push 100.0   ;; b

push 3.9999 ;; a
push 10.0   ;; b

push 0.40999 ;; a
push 1.0     ;; b
```
For example, if `a` is bigger than 0.40999 — 0.41, the last digit of output is correct:
```console
c:\bm>.\build\bin\bme -i .\build\examples\lerp.bm
0.41
0.469
0.528
0.587
0.646
0.705
0.764
0.823
0.882
0.941
1.0
```

**Comment by rexim** [comment]
@kolumb this is really strange! Thank you for reporting that!

**Comment by rexim** [comment]
@kolumb tried to fix the problem in #168. Please check it out when you get a chance and close the issue if the problem is fixed for you.
